File: app/src/main/java/lt/markmerkk/utils/AdvHashSettings.java
package lt.markmerkk.utils;

import java.util.HashMap;
import java.util.Map;
import java.util.Properties;
import lt.markmerkk.Main;

/**
 * Created by mariusmerkevicius on 11/24/15.
 * Stores string data
 */
public class AdvHashSettings extends BaseSettings {
  public static final String PROPERTIES_PATH = "usr.properties";
  Map<String, String> keyValues;

  public AdvHashSettings() {
    keyValues = new HashMap<>();
  }

  public String get(String key) {
    if (!keyValues.containsKey(key))
      return null;
    return keyValues.get(key);
  }

  public void set(String key, String value) {
    if (key == null || value == null) return;
    keyValues.put(key, value);
  }

  @Override
  String propertyPath() {
    return Main.CFG_PATH + PROPERTIES_PATH;
  }

  @Override void onLoad(Properties properties) {
    keyValues.clear();
    for (Object o : properties.keySet())
      if (o instanceof String) {
        String property = properties.getProperty((String) o);
        if (Utils.isEmpty(property)) continue;
        byte[] decodeBytes = Base64.decode(property, 0);
        String decodeValue = new String(decodeBytes);
        keyValues.put((String)o, decodeValue);
      }
  }

  @Override void onSave(Properties properties) {
    for (String s : keyValues.keySet()) {
      String value = keyValues.get(s);
      properties.put(s, new String(Base64.encode(value.getBytes(), 0)).trim());
    }
  }
}

File: app/src/main/java/lt/markmerkk/utils/AdvHashSettings.kt
package lt.markmerkk.utils

import lt.markmerkk.Main
import java.util.*

/**
 * Created by mariusmerkevicius on 11/24/15.
 * Stores string data
 */
class AdvHashSettings : BaseSettings(), HashSettings {

    val keyValues = mutableMapOf<String, String>()

    override operator fun get(key: String): String? {
        if (!keyValues.containsKey(key))
            return null
        return keyValues[key]
    }

    override operator fun set(key: String, value: String) {
        keyValues.put(key, value)
    }

    override fun propertyPath(): String {
        return Main.CFG_PATH + PROPERTIES_PATH
    }

    override fun onLoad(properties: Properties) {
        keyValues.clear()
        for (o in properties.keys)
            if (o is String) {
                val property = properties.getProperty(o)
                if (Utils.isEmpty(property)) continue
                val decodeBytes = Base64.decode(property, 0)
                val decodeValue = String(decodeBytes)
                keyValues.put(o, decodeValue)
            }
    }

    override fun onSave(properties: Properties) {
        for (s in keyValues.keys) {
            val value = keyValues[s]
            properties.put(s, String(Base64.encode(value?.toByteArray(), 0)).trim { it <= ' ' })
        }
    }

    companion object {
        val PROPERTIES_PATH = "usr.properties"
    }
}
